contributed_definitions/nyaml/NXmpes.yaml

```diff
@@ -135,107 +337,345 @@ NXmpes(NXobject):
         exists: optional
         doc: |
           Manipulator for positioning of the sample.
-        sample_temperature(NX_FLOAT):
+        temperature_sensor(NXsensor):
           exists: recommended
-          unit: NX_TEMPERATURE
-        drain_current(NX_FLOAT):
+          name:
+            exists: recommended
+          measurement:
+            enumeration: [temperature]
+          type:
+            exists: optional
+          value(NX_FLOAT):
+            exists: recommended
+        sample_heater(NXactuator):
           exists: recommended
-          unit: NX_CURRENT
-        sample_bias(NX_FLOAT):
+          name:
+            exists: recommended
+          physical_quantity:
+            enumeration: [temperature]
+          type:
+            exists: optional
+          heater_power(NX_FLOAT):
+            exists: recommended
+          (NXpid):
+            exists: recommended
+            setpoint(NX_FLOAT):
+              exists: recommended
+        drain_current_amperemeter(NXsensor):
+          exists: recommended
```

Comment:
optional

Reply:
see above

Reply:
Fine with me. I think this is something few groups do, but we can keep it recommended

Reply:
I agree it's rarely used, so let's just make both `NXsample/drain_current(NXenvironment)` and `NXmanipulator/drain_current_amperemeter` optional, ok?

Reply:
ok